src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Icu.cs

```diff
@@ -50,7 +50,11 @@ private bool InitIcuCultureDataCore()
             index = _sWindowsName.IndexOf(ICU_COLLATION_KEYWORD, StringComparison.Ordinal);
             if (index >= 0)
             {
-                _sName = string.Concat(_sWindowsName.AsSpan(0, index), "_", alternateSortName);
+                // It is possible alternateSortName not set even if the normalized culture name has "@collation=".
+                // "zh-TW-u-co-zhuyin" is a good example of that. The term "u-co-" in the name means the following part will be the sort name.
+                // That is mean "zh-TW-u-co-zhuyin" will have the equivalent ICU name "zh-TW@collation=zhuyin".
+                // In this case, we'll not change the original culture name.
```

Comment:
```suggestion
                // Use original culture name if alternateSortName is not set, which is possible even if the normalized
                // culture name has "@collation=".
                // "zh-TW-u-co-zhuyin" is a good example. The term "u-co-" means the following part will be the sort name
                // and it will be treated in ICU as "zh-TW@collation=zhuyin".
```